src/track/trackmetadatataglib.cpp

```diff
@@ -1507,279 +1522,367 @@ Result readTrackMetadataAndCoverArtFromFile(TrackMetadata* pTrackMetadata, QImag
         pFile = std::move(pOggOpusFile); // transfer ownership
+        break;
+    }
 #endif // TAGLIB_HAS_OPUSFILE
-    } else if (kFileTypeWavPack == fileType) {
+    case FileType::WV:
+    {
         auto pWavPackFile =
-                std::make_unique<TagLib::WavPack::File>(
+                std::make_unique<::TagLib::WavPack::File>(
                         TAGLIB_FILENAME_FROM_QSTRING(fileName));
-        anyTagsWritten |= writeTrackMetadataIntoAPETag(pWavPackFile->APETag(true), trackMetadata);
+        anyTagsModified |= writeTrackMetadataIntoAPETag(pWavPackFile->APETag(true), trackMetadata);
         pFile = std::move(pWavPackFile); // transfer ownership
-    } else if (kFileTypeWAV == fileType) {
+        break;
+    }
+    case FileType::WAV:
+    {
         auto pWAVFile =
-                std::make_unique<TagLib::RIFF::WAV::File>(
+                std::make_unique<::TagLib::RIFF::WAV::File>(
                         TAGLIB_FILENAME_FROM_QSTRING(fileName));
 #if (TAGLIB_HAS_WAV_ID3V2TAG)
-        anyTagsWritten |= writeTrackMetadataIntoID3v2Tag(pWAVFile->ID3v2Tag(), trackMetadata);
+        anyTagsModified |= writeTrackMetadataIntoID3v2Tag(pWAVFile->ID3v2Tag(), trackMetadata);
 #else
-        anyTagsWritten |= writeTrackMetadataIntoID3v2Tag(pWAVFile->tag(), trackMetadata);
+        anyTagsModified |= writeTrackMetadataIntoID3v2Tag(pWAVFile->tag(), trackMetadata);
 #endif
         pFile = std::move(pWAVFile); // transfer ownership
-    } else if (kFileTypeAIFF == fileType) {
+        break;
+    }
+    case FileType::AIFF:
+    {
         auto pAIFFFile =
-                std::make_unique<TagLib::RIFF::AIFF::File>(
+                std::make_unique<::TagLib::RIFF::AIFF::File>(
                         TAGLIB_FILENAME_FROM_QSTRING(fileName));
-        anyTagsWritten |= writeTrackMetadataIntoID3v2Tag(pAIFFFile->tag(), trackMetadata);
+        anyTagsModified |= writeTrackMetadataIntoID3v2Tag(pAIFFFile->tag(), trackMetadata);
         pFile = std::move(pAIFFFile); // transfer ownership
+        break;
     }
-
-    if (!pFile) {
-        qWarning() << "Failed to write track metadata into file" << fileName
-                << ": Unsupported file type" << fileType;
+    default:
+        qWarning()
+            << "Cannot write metadata into file"
+            << fileName
+            << "with unknown or unsupported file type"
+            << fileType;
         return ERR;
     }
-    if (!anyTagsWritten) {
-        qWarning() << "Failed to write track metadata into file" << fileName
-                << ": Inaccessible tags";
+
+    if (!anyTagsModified) {
+        qWarning() << "Failed to modify tags of file" << fileName;
         return ERR;
     }
-    if (!pFile->save()) {
+    // NOTE(uklotzde, 2016-08-23): If pFile == nullptr then writing of tags
+    // using one of the specialized member functions in a derived class
+    // already failed (see above)!
```

Comment:
Done.

On 28.08.2016 00:35, Daniel Schürmann wrote:
>
> In src/track/trackmetadatataglib.cpp 
> <https://github.com/mixxxdj/mixxx/pull/997#discussion_r76523256>:
>
> >          return ERR;
> >      }
> > -    if (!pFile->save()) {
> > +    // NOTE(uklotzde, 2016-08-23): If pFile == nullptr then writing of tags
> > +    // using one of the specialized member functions in a derived class
> > +    // already failed (see above)!
>
> Many things to explain, because the related code is out of sight.
> Can we move it into a function, or call here the right save function via 
> function pointer?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/997/files/c0be58a0f541f274599f28c2bb68826ba5280b4e#r76523256>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ABZ-aoZkZak6Rb3625X0JbfiqBsOLyTLks5qkLutgaJpZM4JustG>.
>